.pipelines/cni/lsg/kernel-upgrade-template.yaml

```diff
@@ -0,0 +1,307 @@
+parameters:
+  dependsOn: ""
+  name: ""
+  clusterType: ""
+  clusterName: ""
+  nodeCount: ""
+  vmSize: ""
+  os: linux
+  arch: ""
+  osSKU: Ubuntu
+
+# CNIv2
+# + Should be able to scale up/down the pods successfully certain number of times.
+# + Node reboot scenarios should be covered.
+# + The CNS state should be validated with that of CNI state.
+# + Pods should have ips assigned and connectivity/datapath test should be present.
+# + CNS restart and validates the state
+
+stages:
+  - stage: create_${{ parameters.name }}
+    variables:
+      commitID: $[ stagedependencies.setup.env.outputs['SetEnvVars.commitID'] ]
+    pool:
+      name: $(BUILD_POOL_NAME_DEFAULT)
+    dependsOn:
+      - setup
+    displayName: "Create Cluster - ${{ parameters.clusterName }}"
+    jobs:
+      - job: create_aks_cluster_with_${{ parameters.name }}
+        steps:
+          - task: AzureCLI@1
+            inputs:
+              azureSubscription: $(BUILD_VALIDATIONS_SERVICE_CONNECTION)
+              scriptLocation: "inlineScript"
+              scriptType: "bash"
+              addSpnToEnvironment: true
+              inlineScript: |
+                make -C ./hack/aks ${{ parameters.clusterType }} \
+                AZCLI=az REGION=$(LOCATION) SUB=$(SUB_AZURE_NETWORK_AGENT_BUILD_VALIDATIONS) \
+                CLUSTER=${{ parameters.clusterName }}-$(commitID) \
+                VM_SIZE=${{ parameters.vmSize }} \
+                AUTOUPGRADE=none
+            name: "CreateAksCluster"
+            displayName: "Create AKS Cluster"
+
+  - stage: ${{ parameters.name }}
+    variables:
+      commitID: $[ stagedependencies.setup.env.outputs['SetEnvVars.commitID'] ]
+      cnsVersion: $[ stagedependencies.setup.env.outputs['SetEnvVars.cnsVersion'] ]
+      cniVersion: $[ stagedependencies.setup.env.outputs['SetEnvVars.cniVersion'] ]
+    pool:
+      name: $(BUILD_POOL_NAME_DEFAULT)
+    dependsOn:
+      - create_${{ parameters.name }}
+      - setup
+    displayName: "CNIv2 Test - ${{ parameters.name }}"
```

Comment:
Are we going to add to this file for cniv1 or other flavors of CNI ?

Reply:
We will be doing this test for Cilium as well